specification/sections/04-hooks.md

```diff
@@ -81,75 +83,154 @@ see: [dynamic-context paradigm](../glossary.md#dynamic-context-paradigm)
 
 > Condition: The provider `metadata` field in the `hook context` **MUST** be immutable.
 
-### 4.3. Hook creation and parameters
+### 4.3. Hook data
```

Comment:
I prefer not renumbering, but I did in this case as conceptually it makes sense to group all the parameter requirements together. I could move this to retain the existing numbering.

Reply:
I think I prefer moving it to avoid the number thrashing. It's slightly inconvenient that things aren't grouped, but I think it's more important to attempt to maintain stability of the requirement numbers.

Reply:
I will update it.